File: src/rs/image.rs
//! Still-image codecs and stateless image transformations.
//!
//! The semantic [`crate::Image`] value is implemented by [`crate::core`].
//! Operations belong here once they have an operation schema, an executable
//! lowering, and an independent correctness oracle.

mod codec;
mod color;
mod filter;
mod geometric;
mod pixel;

pub use codec::{
	ImageCodec, can_decode, can_encode, decode_file, decode_memory, encode, save_file,
	save_rgba_file,
};
pub use color::{
	NormalizationParams, convert_color, normalize, resize_normalize, segmentation_overlay,
};

pub use filter::{
	adaptive_threshold_gaussian, adaptive_threshold_mean, average_blur, bilateral_filter,
	convolve_2d, dilate, erode, gaussian_blur, laplacian, median_blur, morphology_black_hat,
	morphology_close, morphology_gradient, morphology_open, morphology_top_hat, scharr,
	separable_convolve_2d, sharpen, sobel, unsharp_mask,
};

pub use geometric::{
	BorderMode, InterpolationMode, center_crop, crop, flip, pad, remap, resize, rotate,
	warp_affine, warp_perspective,
};
pub use pixel::{
	alpha_blend, brightness_contrast, channel_reorder, clamp, color_twist, composite, erase,
	gamma_contrast, gaussian_noise, grayscale, in_range, invert, posterize, salt_pepper_noise,
	solarize, threshold_binary, threshold_binary_inv, threshold_to_zero, threshold_to_zero_inv,
	threshold_truncate,
};

File: sdk/py/python/oa/image.py
"""Image values, codecs, and stateless transformations."""

from ._native import (
	Image,
	image_brightness_contrast as brightness_contrast,
	image_center_crop as center_crop,
	image_clamp as clamp,
	image_convert_color as convert_color,
	image_crop as crop,
	image_decode_file as decode_file,
	image_decode_memory as decode_memory,
	image_encode as encode,
	image_flip as flip,
	image_gaussian_blur as gaussian_blur,
	image_grayscale as grayscale,
	image_invert as invert,
	image_normalize as normalize,
	image_pad as pad,
	image_resize as resize,
	image_rotate as rotate,
	image_save_file as save_file,
)

__all__ = [
	"Image", "brightness_contrast", "center_crop", "clamp", "convert_color",
	"crop", "decode_file", "decode_memory", "encode", "flip", "gaussian_blur",
	"grayscale", "invert", "normalize", "pad", "resize", "rotate", "save_file",
]
